pkg/config/authorization/opa.go

```diff
@@ -74,24 +78,48 @@ func (opa *OPA) Call(pipeline common.AuthPipeline, ctx context.Context) (bool, e
 		results, err := opa.policy.Eval(opa.opaContext, options)
 
 		if err != nil {
-			return false, err
+			return nil, err
 		} else if len(results) == 0 {
-			return false, fmt.Errorf(invalidOPAResponseErrorMsg)
-		} else if allowed := results[0].Bindings["allowed"].(bool); !allowed {
```

Comment:
Should we check for the key before casting or will it always be there?

Reply:
> Should we check for the key before casting or will it always be there?

It will always be there:

https://github.com/Kuadrant/authorino/blob/65c2751473ddeaf1af6fda7e0e895c1d9c60e040/pkg/config/authorization/opa.go#L21

> what would happen if I wrote a rule called allowed that returned a set here?

I've tested and... It crashes. Nice catch! Thanks!

> Perhaps we have a test for this case?

Now we do 😜

Thanks again!